python/tests/common.py

```diff
@@ -10,6 +10,16 @@ def let_op(pattern, atom, templ):
     space.add_atom(atom)
     return space.subst(pattern, templ)
 
+def letrec_op(subs, body):
+    # just unsugaring `let*`` into `let` substitution by substitution
+    subs = subs.get_children()
+    if len(subs) == 0:
+        return [body]
+    next_sub = subs[0].get_children()
+    if len(subs) == 1:
+        return [E(letAtom, next_sub[0], next_sub[1], body)]
+    return [E(letAtom, next_sub[0], next_sub[1], E(letrecAtom, E(*subs[1:]), body))]
```

Comment:
will it work with 3 or more subexpressions? There is no test for it

Reply:
Yes, it will. This PR is more about `match` than `let*`. It would be better to introduce `let*` in one PR with unit tests for it, and then to change `match` type in another PR, but `match` is more prioritized.